# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../libx86/lheader.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur libx86-1.1-orig/lrmi.c libx86-1.1/lrmi.c
--- libx86-1.1-orig/lrmi.c	2009-09-29 16:34:06.000000000 +0000
+++ libx86-1.1/lrmi.c	2009-09-29 16:34:52.000000000 +0000
@@ -54,6 +54,13 @@
 #include "lrmi.h"
 #include "x86-common.h"
 
+#if defined(__linux__) && !defined(TF_MASK)
+	#define TF_MASK X86_EFLAGS_TF
+	#define IF_MASK X86_EFLAGS_IF
+	#define VIF_MASK X86_EFLAGS_VIF
+	#define IOPL_MASK X86_EFLAGS_IOPL
+#endif
+
 #if defined(__linux__)
 #define DEFAULT_VM86_FLAGS 	(IF_MASK | IOPL_MASK)
 #elif defined(__NetBSD__) || defined(__FreeBSD__)
